src/adapter/src/subscribe.rs

```diff
@@ -154,44 +165,95 @@ impl ActiveSubscribe {
                                         .collect_vec();
 
                                     // Four cases:
-                                    // [(key, value, +1)] => ("upsert", key, value
-                                    // [(key, v1, -1), (key, v2, +1)] => ("upsert", key, v2)
-                                    // [(key, value, -1)] => ("delete", key, NULL)
-                                    // everything else => ("key_violation", key, NULL)
+                                    // [(key, value, +1)] => ("insert", key, NULL, value)
+                                    // [(key, v1, -1), (key, v2, +1)] => ("upsert", key, v1, v2)
+                                    // [(key, value, -1)] => ("delete", key, value, NULL)
+                                    // everything else => ("key_violation", key, NULL, NULL)
+                                    let value_columns = self.arity - order_by_keys.len();
                                     let mut packer = row_buf.packer();
                                     new_rows.push(match &group[..] {
-                                        [(_, row, 1)] | [(_, _, -1), (_, row, 1)] => {
-                                            // upsert
+                                        [(_, row, 1)] => {
+                                            packer.push(if debezium {
+                                                Datum::String("insert")
+                                            } else {
+                                                UPSERT
+                                            });
                                             let datums = datum_vec.borrow_with(row);
                                             for column_order in order_by_keys {
                                                 packer.push(datums[column_order.column]);
                                             }
+                                            if debezium {
+                                                for _ in 0..value_columns {
+                                                    packer.push(Datum::Null);
+                                                }
+                                            }
                                             for idx in 0..self.arity {
                                                 if !order_by_keys.iter().any(|co| co.column == idx)
                                                 {
                                                     packer.push(datums[idx]);
                                                 }
                                             }
-                                            (start.0, row_buf.clone(), 1)
+                                            (start.0, row_buf.clone(), 0)
```

Comment:
Why did all the diffs get changed to 0?

Reply:
They don't get read below when outputting them so I tried not to make them meaningful

Reply:
If they sometimes don't get used, this should be an `Option`!